### PR TITLE
Update Readme with better structure and missing information

### PR DESCRIPTION
**Pull request by x-m-el**
## Overview
This restructures the whole readme and adds some extra information.
This is just a first iteration. I didn't fully rewrite some parts as I found quite a few bugs and missing info while writing this, which I will fix in separate instances. Otherwise this will become too big and also hard to follow with merging every time. 

##### connected issues and PRs:
- https://binnenland.atlassian.net/browse/GN-4513 (should be good enough now)
- https://binnenland.atlassian.net/browse/GN-4515 (at least a good start)
- https://binnenland.atlassian.net/browse/GN-4512

### How to test/reproduce
The plugins and developer docs is the most important part to check out. These should be clear enough to understand for someone not familiar with the original editor.

### Challenges/uncertainties
This documentation is written as how it is *now*. Some things will change because of fixes. Those parts I didn't put too much effort in.

There are quite a few things that still have to be done, but will be done later because they are connected to other issues
- refer to the Dutch name of everything (like buttons) when the translation of plugins is fixed (currently everything stays in English irregardless of browser language)
- Pictures are hosted on imgur and are quite bad. But as we want to focus on Dutch speaking people, I want to change those when the buttons are in Dutch too.
- I didn't change the API documentation, because currently the code seems to have some peculiarities (editor.controller is undefined, on/off method don't exist anymore I think).

After some of the bugfixes, I'll need to touch up the readme to follow these changes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -126,153 +133,189 @@ editorElement.initEditor(arrayOfPluginNames, userConfigObject);
...
+
+#### editorElement API
+These are function available from the editor element, which is the HTML element with the class `notule-editor`. 
+- `initEditor(arrayOfPluginNames: string[], userConfigObject)`: Initialize the editor by passing an array of plugin names that should be activated and an object that contains the configuration for the editor and its plugins. See [Managing Plugins](managing-plugins) for more info. 
+  :warning: **This has to be called before accessing any other methods**.
+- `enableEnvironmentBanner()`: enable the banner that shows the environment and versions of plugins used.
+- `disableEnvironmentBanner()`: disable the banner.
+- `controller`: direct access the the [SayController](https://github.com/lblod/ember-rdfa-editor/blob/d4472d2e237256d30333cfcc20ce6eea7db241f2/addon/core/say-controller.ts) object. See [controller API](controller-api).
```
Comment (elpoelma) [inline README.md:167]:
```suggestion
- `controller`: provides direct access to a [SayController](https://github.com/lblod/ember-rdfa-editor/blob/d4472d2e237256d30333cfcc20ce6eea7db241f2/addon/core/say-controller.ts) object. See [controller API](controller-api).
```

File: README.md [hunk 2]
```diff
@@ -126,153 +133,189 @@ editorElement.initEditor(arrayOfPluginNames, userConfigObject);
...
+- `setHtmlContent(content: string)`: set the HTML content inside the editor, overwriting all previous content.  
+- `getHtmlContent()`: Get the HTML content of the editor. This can be different than custom content set via `setHtmlContent`, because of HTML parsing logic.
```
Comment (elpoelma) [inline README.md:169]:
```suggestion
- `getHtmlContent()`: Get the HTML content of the editor. This might be different than custom content set via `setHtmlContent`, because of HTML parsing logic.
```

File: README.md [hunk 3]
```diff
@@ -126,153 +133,189 @@ editorElement.initEditor(arrayOfPluginNames, userConfigObject);
...
+- `withTransaction(callback: (tr: Transaction) => Transaction | null, includeEmbeddedView = false)`: apply a [Prosemirror transaction](https://prosemirror.net/docs/ref/#state.Transaction) on the main view (or currently active embedded view). When you want to apply the transaction, the callback should return the transaction object.
+- `mainEditorState`: the [editor state](https://prosemirror.net/docs/ref/#state.Editor_State) instance of the main editor
+- `activeEditorState`: the state instance of the active editor. This is usually the same as the `mainEditorState`, except when inside a nested instance (like in a variable field). For typical use, the `mainEditorState` is the one you will most likely need.
+- `mainEditorView`: the [view](https://prosemirror.net/docs/ref/#view.EditorView) instance of the main editor
+- `activeEditorView`: the view instance of the active editor. (see above for the distinction between the main and active editor)
+- `setActiveView(view: RdfaEditorView)`: activate a specific view.
```
Comment (elpoelma) [inline README.md:184]: I'm not sure if this method needs to be documented for embeddable, as it has quite a specific use case for embedded editors.

File: README.md [hunk 4]
```diff
@@ -126,153 +133,189 @@ editorElement.initEditor(arrayOfPluginNames, userConfigObject);
...
+
+The editor can be customized to best fit your application. 
+* [Managing Plugins](#managing-plugins): A list of plugins you can enable, including explanation of how to use them
+* [Environment banner](#enabling-disabling-the-environment-banner): how to enable/disable this banner
+* [Localization](#localization): language options in the editor
+* [styling](#styling)
+
+## Managing Plugins
+Embeddable ships with the following plugins available. 
+This Readme contains all the important info and configuration for the plugins. For more technical and Ember-specific explanation of every plugin, you can check out the Readme of [lblod/ember-rdfa-editor-lblod-plugins](https://github.com/lblod/ember-rdfa-editor-lblod-plugins).
```
Comment (elpoelma) [inline README.md:196]:
```suggestion
This Readme contains all the important info and configuration for the plugins. For more technical and Ember-specific explanations of every plugin, you can check out the Readme of [lblod/ember-rdfa-editor-lblod-plugins](https://github.com/lblod/ember-rdfa-editor-lblod-plugins).
```

File: README.md [hunk 5]
```diff
@@ -126,153 +133,189 @@ editorElement.initEditor(arrayOfPluginNames, userConfigObject);
...
+
+Every plugin can be enabled by passing its name to `arrayOfPluginNames` array and its configuration to `userConfigObject` with the initialization function `initEditor(arrayOfPluginNames, userConfigObject)`.
```
Comment (elpoelma) [inline README.md:198]:
```suggestion
Every plugin can be enabled by passing its name to the `arrayOfPluginNames` array and its configuration to `userConfigObject` with the initialization function `initEditor(arrayOfPluginNames, userConfigObject)`.
```

File: README.md [hunk 6]
```diff
@@ -126,153 +133,189 @@ editorElement.initEditor(arrayOfPluginNames, userConfigObject);
...
+> :warning: The values shown for the config are default values used if you do not pass a config. If you pass **any** config for a plugin, it will not use any of these defaults and only use the config provided. Make sure to pass all required attributes in this case, even if you do not change them.
+
+* [article-structure](#article-structure): Provides structures to better manage official documents, like titles, chapters, articles and paragraphs. It allows you to insert, move and delete them. It has two modes: `besluit` for besluit articles and `regulatoryStatement` for all other structures.
+* [besluit](#besluit): Provides the correct rdfa-structure for constructing a decisions ("besluiten") and some basic validation if mandatory structures are present in the document.
```
Comment (elpoelma) [inline README.md:202]:
```suggestion
* [besluit](#besluit): Provides the correct rdfa-structure for constructing a decision ("besluiten") and some basic validation which checks whether mandatory structures are present in the document.
```

File: README.md [hunk 7]
```diff
@@ -126,153 +133,189 @@ editorElement.initEditor(arrayOfPluginNames, userConfigObject);
...
+* [citation](#citation): Search and insert references to citations (a legal resource/expression)
+* [rdfa-date](#rdfa-date): Inserting and modify annotated date and times
```
Comment (elpoelma) [inline README.md:204]:
```suggestion
* [rdfa-date](#rdfa-date): Inserting and modifying annotated dates and times
```

File: README.md [hunk 8]
```diff
@@ -126,153 +133,189 @@ editorElement.initEditor(arrayOfPluginNames, userConfigObject);
...
+* [roadsign-regulation](#roadsign-regulation): Insert roadsign regulation, based on the registry managed and provided by MOW (Mobiliteit en Openbare Werken)
```
Comment (elpoelma) [inline README.md:205]:
```suggestion
* [roadsign-regulation](#roadsign-regulation): Insert a roadsign regulation, based on the registry managed and provided by MOW (Mobiliteit en Openbare Werken)
```

File: README.md [hunk 9]
```diff
@@ -126,153 +133,189 @@ editorElement.initEditor(arrayOfPluginNames, userConfigObject);
...
+* [variable](#rdfa-variables): Allows insertion and filling in of custom rdfa variables
+* [formatting-toggle](#formatting-toggle): Allows to toggle the formatting marks with a button
+* [rdfa-blocks-toggle](#rdfa-blocks-toggle): Allows to toggle the visual indications of the rdfa blocks with a button.
+* [template-comments](template-comments): Allows insertion and editing of comment blocks to provide extra information to a user filling in a document. These are visually distinct units with a special RDFa type, which allows them to be filtered out during postprocessing.
+##### General Config options
+There are some options you can pass to `pluginsConfig` in `initEditor` that are not connected to a plugin.
+- `docContent: 'block+'`: The property docContent specifies which nodes are allowed in the document. By default we allow one or more nodes of the supertype block, which includes most content. For more info about this check the [Prosemirror docs](https://prosemirror.net/docs/guide/#schema.content_expressions). 
+  See `public/test.html` where `docContent` is specified to allow [article-structure](#Article Structure) nodes in a specific order.
```
Comment (elpoelma) [inline README.md:214]:
```suggestion
  See `public/test.html` where `docContent` is specified to allow [article-structure](#article-structure) nodes in a specific order.
```

File: README.md [hunk 10]
```diff
@@ -126,153 +133,189 @@ editorElement.initEditor(arrayOfPluginNames, userConfigObject);
...
+This plugin is in charge of inserting and manipulating structures. There are insertion buttons in the insert menu of the right sidebar. The plugin has two modes, being `besluit` and `regulatoryStatement`. 
 
-- `setActiveView(view: RdfaEditorView)`: activate a specific view.
+- `besluit` Mode
+	- Only allowed to insert articles
+- `regulatoryStatement` mode
+	- Able to insert titles, chapters, sections... 
+	- [table-of-contents](#Table of Contents) default config works with this mode.
 
+After inserting a structure and selecting it, a card will show to move and delete the structure. These might be disabled if the action is not possible.
```
Comment (elpoelma) [inline README.md:225]:
```suggestion
After inserting a structure and selecting it, a card will show options to move and delete the structure. These might be disabled if the action is not possible. 
```

File: README.md [hunk 11]
```diff
@@ -126,153 +133,189 @@ editorElement.initEditor(arrayOfPluginNames, userConfigObject);
...
 
-## Configuring the editor
+### Besluit 
+ :warning: The besluit plugin is incompatible with the `regulatoryStatement` mode of  [article-structure](#Article Structure).
 
-The editor can be customized to best fit your application. In order to use the editor with these options, be sure to rebuild the sources.
+This will add needed rdfa structures to create a besluit and add some validation. There is no direct interaction with the plugin. The validation will pop up in the sidebar. You can try this out by deleting the title and seeing the following pop up:
+![besluit plugin](https://imgur.com/iwCudJy.png)
+***
+:heavy_plus_sign: Enable by adding `"besluit"` to `arrayOfPluginNames` array.
```
Comment (elpoelma) [inline README.md:248]:
```suggestion
:heavy_plus_sign: Enable by adding `"besluit"` to the `arrayOfPluginNames` array.
```

File: README.md [hunk 12]
```diff
@@ -126,153 +133,189 @@ editorElement.initEditor(arrayOfPluginNames, userConfigObject);
...
-* `table-of-contents`: Provides a table of contents that allow you to click on it to go to the different sections specified with the article-structure plugin, see more at the [plugin docs](https://github.com/lblod/ember-rdfa-editor-lblod-plugins#table-of-contents-plugin).
-* `formatting-toggle`: Allows to toggle on and off the formatting marks.
-* `rdfa-blocks-toggle`: Allows to toggle on and off the visual indications of the rdfa blocks.
-* `template-comments`: Allows insertion and editing of comment blocks. These are visually distinct units with a special RDFa type, which allows them to be filtered out by postprocessing. These are intended to display information to document creators/editors that is not intended to be included in the final publication. See more at the [plugin docs](https://github.com/lblod/ember-rdfa-editor-lblod-plugins#template-comments-plugin). This plugin has no additional configuration parameters.
+### Citation
+Add the possibility to add references to specific legal documents. There are two ways to use this plugin
 
-See above for how these plugins can be enabled.
+**A. Insert Button**
+Click `insert reference` button in the right sidebar
```
Comment (elpoelma) [inline README.md:255]:
```suggestion
Click the `insert reference` button in the right sidebar.
```

File: README.md [hunk 13]
```diff
@@ -126,153 +133,189 @@ editorElement.initEditor(arrayOfPluginNames, userConfigObject);
...
+This will open a modal where you can search for different type of legal documents, preview them and insert if desired.
```
Comment (elpoelma) [inline README.md:256]:
```suggestion
This will open a modal where you can search for different types of legal documents, preview them and insert them if desired.
```

File: README.md [hunk 14]
```diff
@@ -126,153 +133,189 @@ editorElement.initEditor(arrayOfPluginNames, userConfigObject);
...
+* **koninklijk besluit** [words to search for]
+* **ministerieel besluit** [words to search for]
+* **genummerd besluit** [words to search for]
 
-### Default configuration
-We provide the following defaults in case you enable a plugin and don't provide any configuration to it, you can take it as a base for your desired configuration. Take into account that if you provide any configuration to a plugin all of the default will be overrided, so make sure you include all the relevant attributes.
+After typing this trigger phrase, a card will shop up in the right sidebar with the type and search term filled in. Press `Advanced search` to pop open the same modal as shown in **A.** 
+![citation plugin](https://imgur.com/oerd9rV.png)
+***
+:heavy_plus_sign: Enable by adding `"citation"` to `arrayOfPluginNames` array.
```
Comment (elpoelma) [inline README.md:277]:
```suggestion
:heavy_plus_sign: Enable by adding `"citation"` to the `arrayOfPluginNames` array.
```

File: README.md [hunk 15]
```diff
@@ -126,153 +133,189 @@ editorElement.initEditor(arrayOfPluginNames, userConfigObject);
...
+		- `activeInNodeTypes`: given the Prosemirror schema and editor state, return a `Set` of nodetypes inside which the plugin should be active. Embeddable does not expose the schema directly, so some internal knowledge is needed to use this effectively.
+	- if type `'ranges'`: 
+		- `activeInRanges`: given the Prosemirror editor state, return an array of ranges for the plugin to be active in, for example `[[0,50], [70,100]]`
+
+Both examples show how to activate the plugin for the *whole* document.
+
+### RDFa Date
+A simple plugin to insert and modify *semantic* dates and timestamps in an editor document. When added, an `insert date` button will be available in the right insert sidebar. When selecting a date, a card will show up at the same place to edit this date and choose a format for display. 
+***
+:heavy_plus_sign: Enable by adding `"rdfa-date"` to `arrayOfPluginNames` array.
```
Comment (elpoelma) [inline README.md:312]:
```suggestion
:heavy_plus_sign: Enable by adding `"rdfa-date"` to the `arrayOfPluginNames` array.
```

File: README.md [hunk 16]
```diff
@@ -291,158 +334,170 @@ date: {
...
-articleStructure: {
-  mode: 'besluit',
-}
-```
-This block configures the article-structure plugin. The article-structure plugin has 2 modes:
+- `placeholder`:
+	- `insertDate`: placeholder to show when inserting a date that has no data yet.
+	- `insertDateTime`: placeholder to show when inserting a datetime that has no data yet.
+- `formats`: specify default formats to show for selection in the date card.
+	- `label` (optional): The label shown to the user on the card. If not provided, the format is used instead e.g.: `dd/MMvariable/yyyy`
```
Comment (elpoelma) [inline README.md:341]: What is `variable` doing here?

File: README.md [hunk 17]
```diff
@@ -291,158 +334,170 @@ date: {
...
+	- `key`: A **unique** identifier used for identification in the internal code. 
+	- `dateFormat`: The date format used when this is selected.
+	- `dateTimeFormat`: The datetime format to use when this is selected. Used when the user selected "Include time".
```
Comment (elpoelma) [inline README.md:344]:
```suggestion
	- `dateTimeFormat`: The datetime format to use when this is selected. Used when the user selects "Include time".
```

File: README.md [hunk 18]
```diff
@@ -291,158 +334,170 @@ date: {
...
-}
-```
-This block configures the roadsign-regulation plugin.
-The plugin exposes 2 variables to configure, the first one specifies the endpoint to fetch the roadsigns, and the second one is a way to fix the images that don't specify a base url, in this case `https://register.mobiliteit.vlaanderen.be/` will be used.
+:warning: This plugin will only activate in *besluiten* with a certain rdf type.
+<details><summary>Exhaustive list of types</summary>
 
-### Enabling/disabling the environment banner
-The environment banner is a visual indication of the environment you are currently using and which versions of embeddable, the editor and editor-plugins are in use.
+`https://data.vlaanderen.be/id/concept/BesluitType/4d8f678a-6fa4-4d5f-a2a1-80974e43bf34`
```
Comment (elpoelma) [inline README.md:355]: Maybe the types in this list should be converted to links?

File: README.md [hunk 19]
```diff
@@ -291,158 +334,170 @@ date: {
...
-```javascript
-roadsignRegulation: {
-  endpoint: 'https://dev.roadsigns.lblod.info/sparql',
-  imageBaseUrl: 'https://register.mobiliteit.vlaanderen.be/',
-}
-```
-This block configures the roadsign-regulation plugin.
-The plugin exposes 2 variables to configure, the first one specifies the endpoint to fetch the roadsigns, and the second one is a way to fix the images that don't specify a base url, in this case `https://register.mobiliteit.vlaanderen.be/` will be used.
+:warning: This plugin will only activate in *besluiten* with a certain rdf type.
+<details><summary>Exhaustive list of types</summary>
```
Comment (elpoelma) [inline README.md:353]:
```suggestion
<details><summary>Exhaustive list of decision types in which this plugin will activate</summary>
```

File: README.md [hunk 20]
```diff
@@ -291,158 +334,170 @@ date: {
...
+</details>
 
-### Besluit
-As said above this plugin is mostly useful for validation, if you are working with decisions we recommend to enable this plugin. The only way to interact with this plugin is by deleting the title of a besluit, in this case a card will appear saying that the decision is missing a title and will provide a button to insert it.
+When the cursor is inside such a *besluit*, the button `Voeg mobiliteitsmaatregel in` will appear under the insert menu. Clicking this will show a modal to filter and select roadsign regulation to insert.
 
-![besluit plugin](https://imgur.com/iwCudJy.png)
+![roadsign regulation modal](https://i.imgur.com/z7My8lm.png)
+***
+:heavy_plus_sign: Enable by adding `"roadsign-regulation"` to `arrayOfPluginNames` array.
```
Comment (elpoelma) [inline README.md:374]:
```suggestion
:heavy_plus_sign: Enable by adding `"roadsign-regulation"` to the `arrayOfPluginNames` array.
```

File: README.md [hunk 21]
```diff
@@ -291,158 +334,170 @@ date: {
...
-### Roadsign Regulation
-For this plugin you will need to be in a besluit with one of these types:
-* `https://data.vlaanderen.be/id/concept/BesluitType/4d8f678a-6fa4-4d5f-a2a1-80974e43bf34`
-* `https://data.vlaanderen.be/id/concept/BesluitType/7d95fd2e-3cc9-4a4c-a58e-0fbc408c2f9b`
-* `https://data.vlaanderen.be/id/concept/BesluitType/3bba9f10-faff-49a6-acaa-85af7f2199a3`
-* `https://data.vlaanderen.be/id/concept/BesluitType/0d1278af-b69e-4152-a418-ec5cfd1c7d0b`
-* `https://data.vlaanderen.be/id/concept/BesluitType/e8afe7c5-9640-4db8-8f74-3f023bec3241`
-* `https://data.vlaanderen.be/id/concept/BesluitType/256bd04a-b74b-4f2a-8f5d-14dda4765af9`
-* `https://data.vlaanderen.be/id/concept/BesluitType/67378dd0-5413-474b-8996-d992ef81637a`
+:warning: For use in different situations, open an issue on this repo with the usecase, so we can help. The [Prosemirror schema](https://prosemirror.net/docs/guide/#schema) that is used in the config is public-facing yet, so changing this is not trivial.
```
Comment (elpoelma) [inline README.md:391]:
```suggestion
:warning: For use in different situations, open an issue on this repo with the usecase, so we can help. The [Prosemirror schema](https://prosemirror.net/docs/guide/#schema) that is used in the config is not public-facing yet, so changing this is not trivial.
```

File: README.md [hunk 22]
```diff
@@ -291,158 +334,170 @@ date: {
...
+***
+:heavy_plus_sign: Enable by adding `"table-of-contents"` to `arrayOfPluginNames` array.
```
Comment (elpoelma) [inline README.md:393]:
```suggestion
:heavy_plus_sign: Enable by adding `"table-of-contents"` to the `arrayOfPluginNames` array.
```

File: README.md [hunk 23]
```diff
@@ -291,158 +334,170 @@ date: {
...
+**Types of variables:**
+- *text*: a variable that any text can be typed in
+- *number*: pops up an input box that will validate constraints and includes a button to show the number in words. Constraints (min/max) can be set when inserting the variable.
+- *date*: works just like the [RDFa date plugin](rdfa-date), where a user can input a date in specified formats.
+- *location*: choose out of a list of location options, that can contain placeholders themselves. 
+- *codelist*: when inserting, a specific codelist has to be chosen. This codelist is a list of values the user can choose from to fill in the variable. Either the user can select one (single selection) or multiple (multiple selection). 
+- *address*: when inserted, the user can click this to get a modal for searching addresses from the Belgium address register. This can be used to insert existing addresses.
+  **note**: when searching for submunicipalities, only the main municipality will show up in the search. However, when searching for a street, the correct zip-code will be used.
+***
+:heavy_plus_sign: Enable by adding `"variable"` to `arrayOfPluginNames` array.
```
Comment (elpoelma) [inline README.md:431]:
```suggestion
:heavy_plus_sign: Enable by adding `"variable"` to the `arrayOfPluginNames` array.
```

File: README.md [hunk 24]
```diff
@@ -291,158 +334,170 @@ date: {
...
+		- `zonalLocationCodelistUri`: the URI to search for if the location variable is included in a zonal traffic measure.
+		- `nonZonalLocationCodelistUri`: the URI to search for if the location variable is included in a non-zonal traffic measure.
 
 ### Formatting Toggle
-This plugin provides a toggle in the top bar that says `Show formatting marks`, after this toggle is active all the formatting marks of the document such as break lines, paragraphs... will be visible.
-
+This will add a button in the top toolbar `Show formatting marks`. This toggles the visibility of all formatting marks of the document such as break lines, paragraphs...
 ![document with formatting annotations](https://imgur.com/KTNxuBW.png)
+***
+:heavy_plus_sign: Enable by adding `"formatting-toggle"` to `arrayOfPluginNames` array.
```
Comment (elpoelma) [inline README.md:469]:
```suggestion
:heavy_plus_sign: Enable by adding `"formatting-toggle"` to the `arrayOfPluginNames` array.
```

File: README.md [hunk 25]
```diff
@@ -291,158 +334,170 @@ date: {
...
 
 ### Rdfa Blocks Toggle
-This plugin provides a toggle in the top bar that says `Show annotations`, when you click on this toggle the styling will change to show all the rdfa information contained in the document. This plugin is useful if you want to check for errors in the rdfa structure, or simply have a look at what data the editor is generating behind the scenes.
-
-It is very much a work in progress visualization, meant for expert users.
+This will add a button in the top toolbar `Show annotations`. This toggles the visibility of RDFa information contained in the document. This is useful if you want to check for errors in the RDFa structure, or simply have a look at what data the editor is generating behind the scenes. As such, it is mostly useful for expert users.
 
 ![document with rdfa blocks visible](https://imgur.com/Asdu2aN.png)
+***
+:heavy_plus_sign: Enable by adding `"rdfa-blocks-toggle"` to `arrayOfPluginNames` array.
```
Comment (elpoelma) [inline README.md:477]:
```suggestion
:heavy_plus_sign: Enable by adding `"rdfa-blocks-toggle"` to the `arrayOfPluginNames` array.
```

File: README.md [hunk 26]
```diff
@@ -452,42 +507,43 @@ You will need the following things properly installed on your computer.
...
 
 ### Building
-
 * `ember build` (development)
 * `ember build --environment production` (production)
+### Releasing
+Take special care when releasing a new version of this.
+- The changelog is automatically created using the PRs. 
+- To avoid a connection to Github, build files are not automatically added to the release
+	- Add the files under `dist/assets` from a production build to the release in a .zip archive.
```
Comment (elpoelma) [inline README.md:529]: I think it should be all files under `dist`?

File: README.md [hunk 27]
```diff
@@ -452,42 +507,43 @@ You will need the following things properly installed on your computer.
...
+- To avoid a connection to Github, build files are not automatically added to the release
+	- Add the files under `dist/assets` from a production build to the release in a .zip archive. 
 
-### Deploying
+### Developing
+#### How it works
+This repository includes [the editor](https://github.com/lblod/ember-rdfa-editor) and [editor plugins](https://github.com/lblod/ember-rdfa-editor-lblod-plugins) packaged together with Ember so the editor can be used in projects outside of Ember. This is mostly done by adjust the build process in `ember-cli-build.js`, by specifying the output filename and limiting the chunks to one file, for easy importing.
 
-Specify what it takes to deploy your app.
+the consumer loads the editor in their own div element. This editor is fully defined in `app/components/simple-editor.js`, with consumer-facing logic bound in `insertedInDom` and  logic that needs a controller bound in `handleRdfaEditorInit`.
```
Comment (elpoelma) [inline README.md:535]:
```suggestion
The consumer loads the editor in their own div element. This editor is fully defined in `app/components/simple-editor.js`, with consumer-facing logic bound in `insertedInDom` and  logic that needs a controller bound in `handleRdfaEditorInit`. 
```

**Review by elpoelma — CHANGES_REQUESTED** [review]
Very nice improvements :smile: . Just left some comments related to typos and content.

File: README.md (hunk 26 above)
Comment (x-m-el) [reply README.md:530]: I'm actually not sure. Are the other files referenced in the assets files, and that's the reason they are needed?
It is not really specified in the readme (now or before) if you should also "host" those in the same place, or if only the ones under `assets` are needed and the rest can be ignored.

Either way (if they are needed or not needed), it should be more explicit, so definitely needs an update around that.

File: README.md (hunk 3 above)
Comment (x-m-el) [reply README.md:184]: I think the reason is that if they want to use custom commands, it might work differently/weirdly for variables and such. But I am also not sure if there are many cases where a command they'll write will really need to know the difference.
Maybe leaving it out but instead add a note at the bottom about it is a better idea?

File: README.md (hunk 16 above)
Comment (x-m-el) [reply README.md:341]: middle-click pasting accident :(

File: README.md (hunk 18 above)
Comment (x-m-el) [reply README.md:355]: Right, I didn't think the links would actually go to something (as they don't always...)
but in this case they do, so links make more sense.

File: README.md (hunk 3 above)
Comment (elpoelma) [reply README.md:184]: Yes, maybe or in a section on custom prosemirror commands.

File: README.md (hunk 26 above)
Comment (elpoelma) [reply README.md:530]: In the previous releases, we always bundled the whole `dist` folder I think. But I don't know if the files outside of `assets` are needed.

Comment (x-m-el) [reply README.md:530]: I did a small test and as far as I can tell they are not needed, everything seems to works as expected. I might be missing something though :shrug:

File: README.md (hunk 3 above)
Comment (x-m-el) [reply README.md]: @elpoelma I added a separate section with some explanation about prosemirror commands, linking to the editor wiki.
The explanation of what the difference is between the main and active editor state should really be part of that wiki instead of trying to explain it here without (the wiki has context to make the explanation easier).

If you could check the new addition if it is okay, that would be great :)

File: README.md (hunk 26 above)
Comment (elpoelma) [reply README.md:532]: ok :) makes sense!

File: README.md (hunk 3 above)
Comment (elpoelma) [reply README.md]: looks great!

**Review by elpoelma — APPROVED** [review]

File: README.md (hunk 26 above)
Comment (x-m-el) [reply README.md]: added a bit more explicit explanation, so there is more consistency between the different ways of using Embeddable.
